rockcraft.yaml

```diff
@@ -0,0 +1,170 @@
+---
+name: sdcore-bess
+base: ubuntu:20.04
+version: '1.3'
+summary: SD-Core BESS
+description: SD-Core BESS
+license: Apache-2.0
+platforms:
+  amd64:
+
+parts:
+  xdp:
+    plugin: autotools
+    source: https://github.com/xdp-project/xdp-tools.git
+    source-tag: v1.2.2
+    build-packages:
+      - clang
+      - gcc-multilib
+      - libelf-dev
+      - libpcap-dev
+      - linux-headers-5.4.0-26
+      - llvm
+    prime:
+      - usr/local/lib/x86_64-linux-gnu/*.so
+      - usr/lib/libxdp*
+
+  cndp:
+    after:
+      - xdp
+    plugin: nil
+    source: https://github.com/CloudNativeDataPlane/cndp.git
+    source-commit: d5ce4b9edc2e7ddb46a61b395deffafaf11a0500
+    build-environment:
+      - CNE_DEST_DIR: "${CRAFT_PART_INSTALL}"
+    build-packages:
+      - clang
+      - golang
+      - libbpf-dev
+      - libbsd-dev
+      - libnl-3-dev
+      - libnl-cli-3-dev
+      - libnuma-dev
+      - lld
+    override-build: |
+      make
+      make install
+      make static_build=1 rebuild install
+      pushd lang/go/stats/prometheus/
+      go build -o $CRAFT_PART_INSTALL/prometheus prometheus.go
+    prime:
+      - usr/local/lib/x86_64-linux-gnu/*.so
+      - usr/local/bin/cndpfwd
+
+  dpdk:
+    plugin: meson
+    after:
+      - cndp
+    source: https://fast.dpdk.org/rel/dpdk-20.11.3.tar.gz
+    source-type: tar
+    build-packages:
+      - ca-certificates
+      - libbenchmark-dev
+      - libbpf0
+      - libc-ares-dev
+      - libelf-dev
+      - libgflags2.2
+      - libgoogle-glog-dev
+      - libgrpc++-dev
+      - libgtest-dev
+      - libjson-c-dev
+      - libnuma-dev
+      - libpcap-dev
+      - libprotobuf-dev
+      - libunwind-dev
+      - meson
+      - protobuf-compiler
+      - protobuf-compiler-grpc
+      - python3
+    override-pull: |
+      craftctl default
+      for file in ${CRAFT_PROJECT_DIR}/local/patches/*
+      do
+        patch -i $file -p 1
+      done
+    meson-parameters:
+      - --buildtype=debugoptimized
+      - -Dmachine=x86-64
+    prime:
+      - usr/local/lib/x86_64-linux-gnu/*.so
+
+  bess:
+    plugin: nil
+    after:
+      - dpdk
+    source: https://github.com/omec-project/bess.git
+    source-tag: dpdk-2011-focal
+    source-type: git
+    build-packages:
+      - ca-certificates
+      - libbenchmark-dev
+      - libbpf0
+      - libc-ares-dev
+      - libelf-dev
+      - libgflags2.2
+      - libgoogle-glog-dev
+      - libgrpc++-dev
+      - libgtest-dev
+      - libjson-c-dev
+      - libnuma-dev
+      - libpcap-dev
+      - libprotobuf-dev
+      - libssl-dev
+      - libunwind-dev
+      - meson
+      - pkg-config
+      - protobuf-compiler
+      - protobuf-compiler-grpc
+      - python3
+    stage-packages:
+      - libbpf0
+      - libgflags2.2
+      - libjson-c4
+      - libgraph-easy-perl
+      - iproute2
+      - iptables
+      - iputils-ping
+      - tcpdump
```

Comment:
Do we need all of those?

Reply:
Those are all referenced in the code. I am certain we require iproute2 and iptables, for the other two I am not sure exactly what features would break if we remove them. I am afraid that removing them now would break features that are not exercised by the simulation.